src/library_gl.js

```diff
@@ -3598,6 +3598,10 @@ var LibraryGL = {
       var log = GLctx.getShaderInfoLog(GL.shaders[shader]);
       if (log === null) log = '(unknown error)';
       {{{ makeSetValue('p', '0', 'log.length + 1', 'i32') }}};
+    } else if (pname == 0x8B88) { // GL_SHADER_SOURCE_LENGTH
+      var source = GLctx.getShaderSource(GL.shaders[shader]);
+      if (source === null) source = '(unknown error)';
+      {{{ makeSetValue('p', '0', 'source.length + 1', 'i32') }}};
```

Comment:
Can you correspond the implementation here with the current implementation of  `glGetShaderSource` (as opposed to the implementation above for `GL_INFO_LOG_LENGTH`)? In particular the function currently is implemented as

```javascript
  glGetShaderSource: function(shader, bufSize, length, source) {
#if GL_ASSERTIONS
    GL.validateGLObjectID(GL.shaders, shader, 'glGetShaderSource', 'shader');
#endif
    var result = GLctx.getShaderSource(GL.shaders[shader]);
    if (!result) return; // If an error occurs, nothing will be written to length or source.
    if (bufSize > 0 && source) {
      var numBytesWrittenExclNull = stringToUTF8(result, source, bufSize);
      if (length) {{{ makeSetValue('length', '0', 'numBytesWrittenExclNull', 'i32') }}};
    } else {
      if (length) {{{ makeSetValue('length', '0', 0, 'i32') }}};
    }
```

which should be conformant to the spec. In particular, reading https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGetShaderiv.xml says that `If no source code exists, 0 is returned.`, but this will not return 0. It would be good to match the outputted length to the implementation of `glGetShaderSource`.